components/resources/library/src/macosMain/kotlin/org/jetbrains/compose/resources/Resource.macos.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.resources
+
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.usePinned
+import platform.Foundation.NSData
+import platform.Foundation.NSFileManager
+import platform.posix.memcpy
+
+@ExperimentalResourceApi
+actual fun resource(path: String): Resource = MacOSResourceImpl(path)
+
+@ExperimentalResourceApi
+private class MacOSResourceImpl(val path: String) : Resource {
+    override suspend fun readBytes(): LoadState<ByteArray> {
+        val currentDirectoryPath = NSFileManager.defaultManager().currentDirectoryPath
+        val contentsAtPath: NSData? = NSFileManager.defaultManager().run {
+            contentsAtPath("$currentDirectoryPath/src/macosMain/resources/$path")
```

Comment:
The hierarchy of source sets can be customized (for example like we have in androidx fork: commonMain -> skikoMain -> nativeMain -> macosMain).

I think for now we can limit the resource directory to be in macosMain or commonMain. Maybe leaving a comment or TODO about custom sourceSets might be helpful in the future.

Reply:
Done